### PR TITLE
Add automated api client generation

### PR DESCRIPTION
**Pull request by konstyefremov**
https://github.com/happy-travel/edo/issues/1530

### CODE REVIEW

File: Api/Services/ApiClients/ApiClientService.cs [hunk 1]
```diff
@@ -39,8 +40,41 @@ public async Task<Result<ApiClientInfo>> GetCurrent(AgentContext agent)
                 HasDirectContractsFilter = settings.AdditionalSearchFilters.HasFlag(SearchFilters.DirectContractsOnly)
             };
         }
-        
-        
+
+
+        public async Task<GeneratedApiClient> GenerateApiClient(AgentContext agent)
```
Comment (DFSko) [inline Api/Services/ApiClients/ApiClientService.cs:45]: look IApiClientManagementService

**Comment by DFSko** [comment]
I guess we need to add abilitiy to user to enter the password manually

**Comment by konstyefremov** [comment]
> I guess we need to add abilitiy to user to enter the password manually

I think it's a good idea to not make such suggestions in PR's, but to discuss them separately in their own time.

**Review by AlexeyShok — APPROVED** [review]

File: Api/Services/Management/ApiClientManagementService.cs [hunk 2]
```diff
@@ -101,8 +102,23 @@ Task Remove(ApiClient client)
             Task<Result> WriteAuditLog(ApiClient _)
                 => _managementAuditService.Write(ManagementEventType.AgentApiClientDelete, new AgentApiClientEventData(agentId, agencyId));
         }
-        
-        
+
+
+        public async Task<Result<ApiClientData>> GenerateApiClient(int agencyId, int agentId)
+        {
+            var name = GenericApiClientName + agencyId;
```
Comment (kirillta) [inline Api/Services/Management/ApiClientManagementService.cs:109]: Probably, we need a space here

**Review by kirillta — COMMENTED** [review]
Do you add an option of changing a password in a separate PR?

**Comment by konstyefremov** [comment]
> Do you add an option of changing a password in a separate PR?

In fact, it's implemented here as it uses Set method. If API client doesn't exist, it will generate a new Api Client. If it does exist, it will generate a new password for this client.

**Review by kirillta — APPROVED** [review]

**Pull request merged**